tests/e2e/onboarding.test.ts

```diff
@@ -131,6 +140,134 @@ test('onboarding with a short code', async ({ page, getOnboardingData }) => {
 	await expect(page).toHaveURL(`/onboarding`)
 })
 
+test('onboarding with GitHub OAuth', async ({ page }) => {
+	const code = `${MOCK_CODE_GITHUB}_${crypto.randomUUID()}_PLAYWRIGHT_TEST`
+
+	await page.route(/\/auth\/github(?!\/callback)/, async (route, request) => {
+		const headers = { ...request.headers(), [MOCK_CODE_GITHUB_HEADER]: code }
+		await route.continue({ headers })
+	})
+
+	const ghUser = await insertGitHubUser(code)!
+
+	await page.goto('/signup')
+	await page.getByRole('button', { name: /signup with github/i }).click()
+
+	await expect(page).toHaveURL(/\/onboarding\/github/)
+	expect(
+		page.getByText(new RegExp(`welcome aboard ${ghUser.primaryEmail}`, 'i')),
+	).toBeVisible()
+
+	// fields are pre-populated for the user
+	const usernameInput = page.getByRole('textbox', { name: /username/i })
+	expect(usernameInput).toHaveValue(
+		// @ts-ignore
+		normalizeUsername(ghUser.profile.login),
+	)
+	expect(page.getByRole('textbox', { name: /^name/i })).toHaveValue(
+		ghUser.profile.name,
+	)
+
+	// button currently in 'idle' (neutral) state and so has got no companion
+	const createAccountButton = page.getByRole('button', {
+		name: /create an account/i,
+	})
+	expect(createAccountButton.getByRole('status')).not.toBeVisible()
+	expect(createAccountButton.getByText('cross')).not.toBeAttached()
+
+	// attempt 1
+	// Username is a string of alphanums and underscores from 3 to 20 chars
+	// long, that will be lowercased upon sumbission. See: app/utils/user-validation.ts:
+	await usernameInput.fill('U$er_name') // $ is invalid char
```

Comment:
I'd suggest we split the happy path and validation tests into separate test cases. 

- completes the github onboarding given valid user credentials
- renders a validation error given invalid username
- etc...

This way test failures will let us know more about the problematic area of the code.

Reply:
I imagined this test to be a real life user journey, hence the layout and steps. But indeed, there are a couple of scenarios mixed up in here: happy registration path, happy account linking path, and struggling with the form path :sweat_smile: 

Please  have a lot at the decomposed version.

Also, I guess that's a nit, but we currently have inconsistent naming of tests (description) in e2e tests. The docs will also have varying examples, like [`test('does something')`](https://playwright.dev/docs/api/class-pageassertions) vs [`test('basic test')`](https://playwright.dev/docs/api/class-testinfo) . We also have both naming approaches in our unit [tests](https://github.com/epicweb-dev/epic-stack/blob/main/app/routes/_auth%2B/auth.%24provider.callback.test.ts#L140). I guess this is a matter of convention on the project.

Reply:
I think splitting them by different behaviors would make the most sense. 

When it comes to test names, I highly suggest [these tips](https://www.epicweb.dev/talks/how-to-write-better-test-names). I find them immensely useful when naming test cases (and also arranging test suites).

Reply:
tq, those are useful tips, I also like `prompts to confirm deletion` better than `account deletion` :)